### DISCUSSION TITLE
Builder - CSV Import Data - Invalid file type error

### DISCUSSION THREAD

**Discussion by applicationuser** [Budibase/budibase#9551] [Help] [answered]
Hello,

Receiving error while importing CSV file in builder, see below screenshot.  No error in browser console.

CSV file with 4 columns 

Even export of row data and import same file (excluding auto columns)  also NOT working.

Budibase app cloud version : 2.3.0

![image](https://user-images.githubusercontent.com/42922833/217054358-453aaf82-5bd6-453e-b4fc-bc4cbff871a5.png)

**Comment by shogunpurple** [comment]
Hi @applicationuser - please try again with the latest version. Are you still seeing this issue?

**Reply by applicationuser** [reply]
Hello @shogunpurple 

Tested with budibase cloud version 2.3.6 , still having issue.

as a workaround had to convert csv data into json format and load

**Comment by AndersonVD** [comment]
Hi, same issue here, I export the own database csv and import again, same error

**Reply by melohagan** [reply]
Hey @AndersonVD 

Would you be able to provide reproduction steps please? Is this for an Internal DB table, or a SQL table?

**Reply by AndersonVD** [reply]
I fixed, just updated my docker (i'm running windows with WSL) and restart the container. After that worked perfetcly.

**Comment by JohannesM007** [comment]
Hi, same issue with Version 2.3.12 free (cloud) with Internal DB table

![grafik](https://user-images.githubusercontent.com/125383022/219046779-7664aaa5-06ab-4efd-b2bc-0932531c4d8d.png)
![grafik](https://user-images.githubusercontent.com/125383022/219047105-178997d9-1621-4302-8242-f23fdef22a62.png)

Cheers 
Johannes

**Reply by applicationuser** [reply]
@JohannesM007 

do you still have issue with csv file import?

**Reply by JohannesM007** [reply]
> @JohannesM007
> 
> do you still have issue with csv file import?

Yes its still the same issue.
I use a local BudibaseDB  Version 2.3.17 in the moment.
If i create a new table with just 2 Colums i can't import via CSV, only json.
[Pats.csv](https://github.com/Budibase/budibase/files/11158151/Pats.csv)
My workaround is to use an online converter to create a json file form csv file. This works fine ;-)
Regards, 
Johannes

**Comment by applicationuser** [comment]
@melohagan 

any update when this issue will be fixed? 

tested now with latest app release 2.4.31 still cannot import data through csv file import.

**Reply by melohagan** [reply]
Hey @applicationuser 

Would you be able to send the CSV file to support@budibase.com so that I can try to reproduce please?

**Reply by applicationuser** [reply]
@melohagan 

I was able to reproduce on new app and table, Here are the steps i followed

- Create a table with 2 columns , data type: text 
- import csv file , response:  Invalid file type 

contents of the csv file: 
"Column1","Column2"
"one","two"
"3","four"

![image](https://user-images.githubusercontent.com/42922833/228409161-734ca544-9fff-46c9-aa4b-df6dceec5d15.png)

sending csv file to support email.

**Comment by applicationuser** [comment]
@melohagan  @shogunpurple 

Additional info to reproduce issue.   Budibase cloud, app version: 2.5.2

Issue occurs on Windows 11 OS/Firefox browser (111.0.1)  : On reviewing network tab, No call to  'api/convert/csvToJson' 

Same file imported on Windows 11 OS/Edge browser (112.0.1722.48 ) : network tab has call to 'api/convert/csvToJson'  with response 
 Next call to 'api/tables/validateExistingTableImport' failed with invalid columns error.  see image below 

looks like table column name to csv header name check is case sensitive ,  updated CSV to use same case column name then upload works fine. 

![image](https://user-images.githubusercontent.com/42922833/233150463-7a60fb8d-85cc-439f-8a89-2c10d95156fa.png)


Hope this helps in fixing the issue.

Thanks,

**Reply by melohagan** [reply]
Hey @applicationuser 

To clarify, if the column names match (case-sensitive) then there are no issues with the import? 

Do you only see the `error: invalid file type` message if the column names do not match the correct case?

**Reply by applicationuser** [reply]
Hello @melohagan 

_To clarify, if the column names match (case-sensitive) then there are no issues with the import?_  Yes. Only works in Edge browser not Firefox.

_Do you only see the error: invalid file type message if the column names do not match the correct case?_ No. Firefox browser has this issue.  even when column names match correct case. 

Working Scenario :  Edge browser , csv header names case sensitive match with table column names. 

Non working scenario : Firefox browser all scenarios.

Thanks,

**Reply by melohagan** [reply]
Thank you very much for the extra info @applicationuser 

I'll explore this further!

**Reply by melohagan** [reply, marked as answer]
Hey @applicationuser 

I don't have access to Windows, but I have a theory on what the problem is. Basically I believe that it relates to this: https://github.com/thoughtbot/paperclip/issues/2170

In our code, we explicitly check for `text/csv` whereas it may be the case that the Firefox/Windows combination interprets the file type as an excel file, or a stream: `application/octet-stream`.

I'm going to create a bug issue and ask the dev team if they have any other ideas, or if anyone has a Windows machine to test on. 
https://github.com/Budibase/budibase/issues/10358

**Reply by applicationuser** [reply]
Thank you @melohagan for looking into it.  just a note though , it used to work in Firefox before. 
Started noticing issue from release version : 2.3.0

**Reply by melohagan** [reply]
I can see a change was made in this area on the 17th of January, so that would make sense. There's a PR out for the fix, so we'll aim to get that merged in and deployed ASAP.
